packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -212,15 +270,20 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
       parent: _currentController,
       curve: Curves.easeIn
     );
+    _currentAnimation.addListener(_onProgressChanged);
 
     // If we start out with a child, have the child appear fully visible instead
     // of animating in.
     if (widget.child != null)
       _currentController.value = 1.0;
+    else
+      _updateGeometryScale(_previousAnimation.value);
   }
 
   @override
   void dispose() {
+    _previousAnimation.removeListener(_onProgressChanged);
+    _currentAnimation.removeListener(_onProgressChanged);
```

Comment:
you don't need to remove these, the dispose methods below will do it for you

Reply:
done